### DISCUSSION TITLE
How to debug loco application using vscode or rustrover?

### DISCUSSION THREAD

**Discussion by yuequan1997** [loco-rs/loco#377] [Q&A]
It's not possible to debug directly with ide, but I believe it should be possible to debug through some kind of linkage

**Comment by kaplanelad** [comment]
What do you mean by debug directly with ide? I use `visual studio code`, and a debugger works for me. 
Can you elaborate more on what you expected to implement in loco?

**Reply by yuequan1997** [reply]
May I ask how you did the debugging, because I started the project the `cargo loco start` way and couldn't do it in vscode!

**Comment by tjyang** [comment]
+1 
Works for beginner like me so far,  mouse hover works to look up create info is query helpful.
![image](https://github.com/loco-rs/loco/assets/378638/e9a1d341-437a-48ad-a721-89f218af4051)

**Comment by LimpidCrypto** [comment]
Here is how I do it with VSCode:
.vscode/tasks.json:
```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "build_project",
            "type": "shell",
            "options": {
                "cwd": "${workspaceFolder}",
                "shell": {
                    "executable": "/bin/bash",
                    "args": [
                        "-c",
                        "-i"
                    ]
                }
            },
            "command": "cargo",
            "args": [
                "build"
            ]
        }
    ]
}
```
launch.json
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "lldb",
            "request": "launch",
            "name": "Debug executable 'lc_api_template-cli'",
            "program": "${workspaceFolder}/target/debug/${workspaceFolderBasename}-cli",
            "preLaunchTask": "build_project",
            "args": [
                "start",
                "-e",
                "development"
            ],
            "cwd": "${workspaceFolder}",
        }
    ]
}
```

**Reply by LimpidCrypto** [reply]
My solution requires that your cargo package name is the same as the project basename
